### PR TITLE
PHP docs block update

### PR DESCRIPTION
**Pull request by DIOHz0r**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | N/A

This PR is for a massive update for the PHP doc blocks, still work in progress.

### CODE REVIEW

File: front/backup.php [hunk 1]
```diff
@@ -212,7 +216,9 @@ function get_def($DB, $table) {
  * @param $DB        DB object
  * @param $dumpFile  dump file
  * @param $duree     max delay before refresh
-**/
+ *
+ * @return bool
```
Comment (trasher) [inline front/backup.php]: To be consistent, use full name such as `boolean` or `integer` instead of `bool` and `int`.

File: inc/alert.class.php [hunk 2]
```diff
@@ -191,11 +191,10 @@ static function dropdownIntegerNever($name, $value, $options = []) {
    /**
     * Does alert exists
     *
-    * @param string  $itemtype (default '')
-    * @param integer $items_id (default '')
-    * @param integer $type     (default '')
-    *
```
Comment (trasher) [inline inc/alert.class.php]: Blank line must be kept, indentation is incorrect.

File: inc/auth.class.php [hunk 3]
```diff
@@ -564,12 +563,13 @@ function addToError($message) {
...
     * Manage use authentication and initialize the session
     *
-    * @param string  $login_name     Login
-    * @param string  $login_password Password
-    * @param boolean $noauto         (false by default)
+    * @param string $login_name Login
+    * @param string $login_password Password
+    * @param boolean $noauto (false by default)
     *
-    * @return boolean (success)
```
Comment (trasher) [inline inc/auth.class.php]: Same here (blank line, indentation)

File: inc/auth.class.php [hunk 4]
```diff
@@ -1168,8 +1168,7 @@ static function redirectIfAuthenticated($redirect = null) {
 
    /** Display refresh button in the user page
     *
-    * @param object $user User object
-    *
+    * @param object|User $user User object
```
Comment (trasher) [inline inc/auth.class.php]: Not agree; just `User` in that case. Plus the blank line.

File: inc/authldap.class.php [hunk 5]
```diff
@@ -158,6 +158,10 @@ function preconfig($type) {
       }
    }
 
+   /**
```
Comment (trasher) [inline inc/authldap.class.php]: What is te point to add an empty documentation block? See also documentation about inherited methods.

File: inc/authldap.class.php [hunk 6]
```diff
@@ -3205,8 +3234,7 @@ static function getNumberOfServers() {
    /**
     * Build LDAP filter
     *
-    * @param resource $authldap AuthLDAP object
-    *
+    * @param AuthLDAP|resource $authldap AuthLDAP object
```
Comment (trasher) [inline inc/authldap.class.php]: A `resource` is not expected

File: inc/budget.class.php [hunk 7]
```diff
@@ -110,7 +110,6 @@ static function displayTabContentForItem(CommonGLPI $item, $tabnum = 1, $withtem
     *     - target for the Form
     *     - withtemplate : template or basic item
     *
-    * @return Nothing (display)
```
Comment (trasher) [inline inc/budget.class.php]: All methods exepted constructor must hav ea return type documented. `@return void` should be used here.

File: inc/calendar.class.php [hunk 8]
```diff
@@ -392,12 +394,11 @@ function isAWorkingHour($time) {
     *
     * @param $start              datetime    begin
     * @param $delay              timestamp   delay to add
-    * @param $additional_delay   timestamp   delay to add (default 0)
+    * @param int|timestamp $additional_delay timestamp   delay to add (default 0)
```
Comment (trasher) [inline inc/calendar.class.php]: for me, timestamp is not a type, we really expect an integer here

**Comment by DIOHz0r** [comment]
Hi,

A lot of doc blocks has been updated so l'm going to leave this changes at this point cause there are a lot of stuff to review.

comments are welcome.

**Comment by trasher** [comment]
This is still not OK:
- CI is failing,
- some docblocks does not follow documentation,
- several types are wrong (were correct in the original version),
- ...

Also this is too huge to be reviewed, and this should probably not target the bugfixes branch.

**Pull request closed without merging**